File: backend/entity/venue.go
package entity
import"gorm.io/gorm"

type Venue struct{
	gorm.Model
	VenueName string `json:"venue_name"`
	Location string `json:"location"`
	VenueCapacity int `json:"venue_capacity"`
	
	VenueTypeID uint `json:"venue_type_id"`
	VenueType *VenueType `gorm:"foreignKey: venue_type_id" json:"venue_type"`
	Stages        []Stage     `gorm:"foreignKey:VenueID" json:"stages"` // เพิ่มเพื่อเก็บ Stage(s

}

File: frontend/src/interfaces/venue.ts
import type { StageInterface } from "./stage";
import type { VenueTypeInterface } from "./venuetype";

export interface VenueInterface {
 ID: number;
 
venue_id:number;
  venue_name: string;
  location: string;
  venue_capacity: number;
  venue_type_id: number;
  
  venue_type?: VenueTypeInterface;
  stages?: StageInterface[];
  


}
